app/javascript/packages/document-capture/hooks/use-cookie.js

```diff
@@ -1,50 +1,31 @@
-import { useState, useEffect } from 'react';
+import useForceRender from './use-force-render';
 
 /**
  * React hook to access and manage a cookie value by name.
  *
  * @param {string} name Cookie name.
  *
- * @return {[value: string|null, setValue: (nextValue: string?) => void]}
+ * @return {[getValue: () => string|null, setValue: (nextValue: string?) => void]}
  */
 function useCookie(name) {
-  const getCookieValue = () =>
+  const forceRender = useForceRender();
+
+  const getValue = () =>
     document.cookie
       .split(';')
       .map((part) => part.trim().split('='))
       .find(([key]) => key === name)?.[1] ?? null;
 
-  const [value, setStateValue] = useState(getCookieValue);
-
   /**
    * @param {string?} nextValue Value to set, or null to delete the value.
    */
   function setValue(nextValue) {
     const cookieValue = nextValue === null ? '; Max-Age=0' : nextValue;
     document.cookie = `${name}=${cookieValue}`;
-    setStateValue(nextValue);
```

Comment:
One might argue that we could just keep this in state, since internally calling `setValue` would trigger a re-render. The problem with this is that the value could fall out of sync with multiple instances of the hook (aside: probably want a test case for this). The getter isn't prone to this, since it'll get the value at the time that it's called.

Alternatively, we could introduce some simple shared context which makes sure that if the cookie value is updated, all instances of the hook are re-rendered with the new value.

Reply:
This seems pretty straightforward, I think it's clearer than using a shared context

Reply:
> The problem with this is that the value could fall out of sync with multiple instances of the hook (aside: probably want a test case for this).

Added in cd4d91f3d.

> This seems pretty straightforward, I think it's clearer than using a shared context

I was curious, so I tried in 1f493d566 to see what this might look like. It's a bit more complex, yeah, though not too bad I think? I guess I'm mostly concerned that the hook may give a false sense that things are kept well in-sync.

Reply:
> I guess I'm mostly concerned that the hook may give a false sense that things are kept well in-sync.

Yeah that's my main reason for thinking the getter function is better, because it is the closest thing to a "live" value

Reply:
> Yeah that's my main reason for thinking the getter function is better, because it is the closest thing to a "live" value

Problem is that it's live at the time that it's called, but in a case where we use the value as part of the render logic of a component, only the component instance which calls the setter would trigger an re-render, and any other component instance would continue showing a stale value.

Reply:
oh right I forget how "lazy" react is 😩 happy to trust your judgement on what is worthwhile vs overkill for something like this

Reply:
I wish it were easier to manage a "global" value than juggling subscribers like this, but I think it's fine enough. I'd also like to limit our use of "force render" vs. built-in alternatives like `setState`, so I feel a bit happier with this approach.